src/uct/tcp/tcp_cm.c

```diff
@@ -326,8 +372,9 @@ void uct_tcp_cm_insert_ep(uct_tcp_iface_t *iface, uct_tcp_ep_t *ep)
                (ctx_caps == UCT_TCP_EP_FLAG_CTX_TYPE_RX));
     ucs_assert(!(ep->flags & UCT_TCP_EP_FLAG_ON_MATCH_CTX));
 
-    ucs_conn_match_insert(&iface->conn_match_ctx, &ep->peer_addr,
-                          ep->conn_sn, &ep->elem,
+    ucs_conn_match_insert(!(ep->flags & UCT_TCP_EP_FLAG_CONNECT_TO_EP) ?
```

Comment:
btw, why are you using `!` in similar cheks? could do just `&`, but change order of results
the same is relevanr for `uct_tcp_cm_ep_set_conn_sn`

Reply:
fixed